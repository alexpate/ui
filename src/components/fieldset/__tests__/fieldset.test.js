/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import Fieldset from '../fieldset';

expect.addSnapshotSerializer(serializer);

describe('components/fieldset', () => {
  describe('Fieldset (default)', () => {
    const create = (spread = {}) => {
      const props = {
        // (insert your default props here)
        ...spread,
      };
      const element = <Fieldset {...props} />;
      return {props, element};
    };

    it('should shallow render as expected', () => {
      const {element} = create({
        // (insert test specific props here)
        children: 'Child',
      });
      const result = shallow(element);
      expect(result).toMatchSnapshot();
    });
  });
});
