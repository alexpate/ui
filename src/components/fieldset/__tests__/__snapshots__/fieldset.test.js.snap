// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render shallow component ok 1`] = `
<fieldset
  className="css-155rrlq0 css-1u5seyf"
  name={undefined}
  onClick={undefined}
  style={Object {}}
>
  <div
    className="css-134evea0 css-1lsspws"
    onClick={undefined}
    style={Object {}}
  >
    <div
      className="css-134evea0 css-1x0f447"
      style={Object {}}
    >
      <label
        className="css-16ef8bd0 css-kzbo9d"
        color="navy600"
        htmlFor={undefined}
        onClick={undefined}
        size="small"
        style={Object {}}
        title={undefined}
      >
        Your name
        <span
          className="css-134evea3 css-14z43x8"
          style={Object {}}
        >
          *
        </span>
      </label>
    </div>
    <div
      className="css-13nj03s1 css-11q3kj7"
      style={Object {}}
    >
      <input
        className="css-kus6en css-13nj03s2"
        defaultValue={undefined}
        name="Alex"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Alex"
        readOnly={undefined}
        size="medium"
        style={
          Object {
            "width": undefined,
          }
        }
        value={undefined}
      />
    </div>
  </div>
</fieldset>
`;
